### PR TITLE
[Fix] Replace value.numpy()[0] with float(value)

### PR DESCRIPTION
**Pull request by juncaipeng**
To support 0 dim Tensor, replace `value.numpy()[0]` with `float(value)` when the shape of value is `[1]`

### CODE REVIEW

File: contrib/DomainAdaptation/script/train.py [hunk 1]
```diff
@@ -177,8 +177,8 @@ def train(self,
                                                      labels_src)
 
                 loss_src_seg = loss_src_seg_main + loss_src_seg_aux
```
Comment (shiyutang) [inline contrib/DomainAdaptation/script/train.py:179]: This file checked

File: contrib/MedicalSeg/medicalseg/core/train.py [hunk 2]
```diff
@@ -161,7 +161,7 @@ def train(model,
             model.clear_gradients()
             # TODO use a function to record, print lossetc
 
-            avg_loss += loss.numpy()[0]
+            avg_loss += float(loss)
             mdice += np.mean(per_channel_dice) * 100
```
Comment (shiyutang) [inline contrib/MedicalSeg/medicalseg/core/train.py:166]: checked

File: contrib/MedicalSeg/nnunet/core/train.py [hunk 3]
```diff
@@ -141,7 +141,7 @@ def train(model,
             train_profiler.add_profiler_step(profiler_options)
 
             model.clear_gradients()
-            avg_loss += loss.numpy()[0]
+            avg_loss += float(loss)
             mdice += np.mean(per_channel_dice) * 100
```
Comment (shiyutang) [inline contrib/MedicalSeg/nnunet/core/train.py:145]: checked

**Review by shiyutang — APPROVED** [review]
LGTM

**Review by wuyefeilin — APPROVED** [review]
LGTM
